nodejs/week3/meal-sharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,151 @@
+const { log } = require("debug");
+const { response, request } = require("express");
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+
+
+/*router.get("/", async (request, response) => {
+  try {
+    // knex syntax for selecting things. Look up the documentation for knex for further info
+    const titles = await knex("meal").select("title");
+    response.json(titles);
+  } catch (error) {
+    throw error;
+  }
+});*/
+
+//  all meals 
+/*router.get("/", async (request, response) => {
+try {
+  const allMeals = await knex("meal").select();
+  response.json(allMeals)
+} catch (error) {
+  throw error;
+}
+});
+// Adds a new meal
+router.post("/", async (request, response) => {
+  try {
+    await knex("meal").insert(request.body)
+    response.json({"success": true})
+  } catch (error) {
+    response.json({"success": false})
+    throw error;
+  }
+});
+// Returns meal by id
+router.get('/:id', async (request, response) => {
+      const meal = await knex("meal").select().where({id: request.params.id});
+      response.json(meal);
+});
+
+// updates the meal by id 
+router.put("/:id", async (request, response) => {
+  const meal = await knex("meal").where({id: request.params.id}).update(request.body);
+  response.json(meal)
+})
+// Delete meal by ID
+router.delete("/:id", async (request, response) => {
+  const meal = await knex("meal").where({id: request.params.id}).del();
+  response.json(meal);
+})*/
+
+
+// GET api/meals/ query parameters and you should make sure that you can combine them
+/*router.get("/", async (request, response) => {
+    if (request.query.maxPrice) {
+      const maxPriceMeal = await knex("meal").select().where('price', '<', Number(request.query.maxPrice))
+      response.json(maxPriceMeal)
+    };
+    if (request.query.availableReservations) {
+      const availableMeals = await knex("meal")
+      .join('reservation', 'mealid', 'reservations.mealid')
+      .select('meal.id', 'title', 'maxreservatins')
+      .where("maxreservation", ">", "numberofguests")
+      response.json(availableMeals)
+    };
+    // a bit dif here just to try som new
+    if ('title' in request.query) {
+      const title = await knex("meal").where(meal.title);
+      const getTitle = meals.find((meal) =>
+      meal.title.toLowerCase().includes(title.toLowerCase()),
+      response.send(getTitle)
+      )};
+
+    if ("createdAfter" in req.query) {
+      const createdAfterMeals = await knex("meal").where("createddate", ">", req.query.createdAfter)
+      response.json(createdAfterMeals)
+    };
+
+    if (limit.request.query) {
+      const limitMeals =  await knex("meal").where.()
+      if (meals.length >= 6) {
+        meals.splice(0, limit)
+      }
+          response.send(limit)
+  });*/
+
+  router.get("/", async (request, response) => {
+    const meals = await knex("meal").select();
```

Comment:
Something more advanced:
Generally, it's a good practice to try and first build your query with knex and then execute it (await operator) instead of first loading your data and then filtering them. That way, in case you have a very large amount of data in your database, you avoid having to load all of them.
Here though, you have so few meals it doesn't matter, just something to have in mind :)
If you do want to improve it though, I suggest you also watch this [video ](https://www.youtube.com/watch?v=4KV79W4g2QU)from Orhan which does what you want to accomplish with a slightly different example.